app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt

```diff
@@ -155,15 +183,23 @@ class AttendeeViewModel(private val attendeeService: AttendeeService, private va
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe({
-                    val order = Order(
-                            id = getId(),
-                            paymentMode = if (it.price == null || it.price <= 0) "free" else paymentOption.toLowerCase(),
-                            country = country,
-                            status = "pending",
-                            amount = if (it.price == null || it.price <= 0) null else it.price,
-                            attendees = arrayListOf(AttendeeId(attendeeId)),
-                            event = EventId(eventId))
-                    createOrder(order)
+                    tickets.value?.add(it)
+                    Timber.d("Loaded tickets! %s", tickets.value?.toList().toString())
+                    if (tickets.value?.size == attendees.value?.size) {
+                        val attendeeList = attendees.value?.map { AttendeeId(it.id) }?.toList()
+                        var amount = totalAmount.value
+                        var paymentMode = paymentOption?.toLowerCase()
+                        if (amount == null || amount <= 0) {
+                            paymentMode = "free"
+                            amount = null
+                        }
+                        val eventId = event.value?.id
+                        if (eventId != null) {
+                            createOrder(Order(getId(), paymentMode, country, "pending", amount, attendees = attendeeList, event = EventId(eventId)))
```

Comment:
Again passing unneeded parameters inside.

This ViewModel has so many problems.